SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SchemaGenerator.cpp

```diff
@@ -303,12 +290,62 @@ void GenerateSubobjectSchema(UClass* Class, TSharedPtr<FUnrealType> TypeInfo, FS
 		Writer.Outdent().Print("}");
 	}
 
+	uint32 DynamicComponentsPerClass = GetDefault<USpatialGDKSettings>()->MaxDynamicallyAttachedSubobjectsPerClass;
+
+	FSubobjectSchemaData SubobjectSchemaData;
+
+	for (uint32 i = 1; i <= DynamicComponentsPerClass; i++)
+	{
+		FDynamicSubobjectSchemaData DynamicSubobjectComponents;
+
+		for (EReplicatedPropertyGroup Group : GetAllReplicatedPropertyGroups())
+		{
+			if (RepData[Group].Num() == 0 && Group == REP_SingleClient)
+			{
+				continue;
+			}
+
+			Writer.PrintNewLine();
+
+			Worker_ComponentId ComponentId = IdGenerator.Next();
+			FString ComponentName = SchemaReplicatedDataName(Group, Class) + TEXT("Dynamic") + FString::FromInt(i);
```

Comment:
Are we checking if I make a schema called "ThingoDynamic1" and then I also have "Thingo" and enable dynamic componenets that we don't have collisions?

Reply:
Put ticket number in.